ttl/ttlworker/session.go

```diff
@@ -57,10 +60,30 @@ func getSession(pool sessionPool) (session.Session, error) {
 		return nil, errors.Errorf("%T cannot be casted to sqlexec.SQLExecutor", sctx)
 	}
 
-	se := session.NewSession(sctx, exec, func() {
+	defaultRetryLimit := int64(variable.DefTiDBRetryLimit)
+	retryLimit := &defaultRetryLimit
```

Comment:
I think you can just get original retry limit form `sessionVars.RetryLimit`

Reply:
Fixed.